kfs-core/src/main/java/edu/arizona/kfs/fp/document/validation/impl/ErrorCertificationValidation.java

```diff
@@ -0,0 +1,61 @@
+package edu.arizona.kfs.fp.document.validation.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.sys.document.validation.GenericValidation;
+import org.kuali.kfs.sys.document.validation.event.AttributedDocumentEvent;
+import org.kuali.rice.krad.util.GlobalVariables;
+
+import edu.arizona.kfs.fp.businessobject.ErrorCertification;
+import edu.arizona.kfs.fp.document.GeneralErrorCorrectionDocument;
+import edu.arizona.kfs.sys.KFSConstants;
+import edu.arizona.kfs.sys.KFSKeyConstants;
+
+public class ErrorCertificationValidation extends GenericValidation {
+	
+	private static int NUM_ERROR_CERT_FIELDS = 4;
+	
+	protected static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ErrorCertificationValidation.class);
+	
+	public boolean validate(AttributedDocumentEvent event) {
+		GeneralErrorCorrectionDocument currDocument = (GeneralErrorCorrectionDocument)event.getDocument();
+		int numBlankFields = errorCertificationBlankFieldsCheck(currDocument);
+		LOG.debug("numBlankFields: " + numBlankFields);
+		
+		//if Error Certification tab is partially filled out
+		if((numBlankFields > 1) && (numBlankFields < KFSConstants.ErrorCertificationConstants.NUM_ERROR_CERT_FIELDS)) {
+			GlobalVariables.getMessageMap().putErrorForSectionId("document.errorCertification", KFSKeyConstants.ERROR_ERROR_CERT_FIELDS_REQ, "error.errorCert.fieldsReq");
+			return false;
+		}
+		
+		return true;
+	}
+
+	/**
+	 * This method goes through all the Error Certification fields looking for blank fields.
+	 * @param document
+	 * @return blankFieldCount the number of blank fields in the tab
+	 */
+	public int errorCertificationBlankFieldsCheck(GeneralErrorCorrectionDocument document) {
```

Comment:
Oh yes, you're right... So yes, threw me off, since it's the convention the vast majority of the time to directly extend the existing class rather than just the interface. My IDE has this method flagged as duplicate code from the kuali package version of this class. And actually, if you had extended the kuali version, you'd get this method for free. Out of curiosity, any reason not to extend the kuali version?